esphome/components/mcp2515/mcp2515.h

```diff
@@ -0,0 +1,119 @@
+  CANINTF_WAKIF = 0x40,
+  CANINTF_MERRF = 0x80
+};
+
+enum /*class*/ EFLG : uint8_t {
+  EFLG_RX1OVR = (1 << 7),
+  EFLG_RX0OVR = (1 << 6),
+  EFLG_TXBO = (1 << 5),
+  EFLG_TXEP = (1 << 4),
+  EFLG_RXEP = (1 << 3),
+  EFLG_TXWAR = (1 << 2),
+  EFLG_RXWAR = (1 << 1),
+  EFLG_EWARN = (1 << 0)
+};
+
+enum /*class*/ STAT : uint8_t { STAT_RX0IF = (1 << 0), STAT_RX1IF = (1 << 1) };
+
+static const uint8_t STAT_RXIF_MASK = STAT_RX0IF | STAT_RX1IF;
+static const uint8_t EFLG_ERRORMASK =
+    EFLG_RX1OVR | EFLG_RX0OVR | EFLG_TXBO | EFLG_TXEP | EFLG_RXEP;
+
+class MCP2515
+    : public canbus::Canbus,
+      public spi::SPIDevice<spi::BIT_ORDER_MSB_FIRST, spi::CLOCK_POLARITY_LOW,
+                            spi::CLOCK_PHASE_LEADING, spi::DATA_RATE_8MHZ> {
+public:
+  MCP2515(){};
+  void set_mcp_clock(CanClock clock) { this->mcp_clock_ = clock; };
+  void set_mcp_mode(const CANCTRL_REQOP_MODE mode) { this->mcp_mode_ = mode; }
+  static const struct TxBnRegs {
+    REGISTER CTRL;
+    REGISTER SIDH;
+    REGISTER DATA;
+  } TXB[N_TXBUFFERS];
+
+  static const struct RxBnRegs {
+    REGISTER CTRL;
+    REGISTER SIDH;
+    REGISTER DATA;
+    CANINTF CANINTF_RXnIF;
+  } RXB[N_RXBUFFERS];
+
+protected:
+  CanClock mcp_clock_{MCP_8MHZ};
+  CANCTRL_REQOP_MODE mcp_mode_ = CANCTRL_REQOP_NORMAL;
+  bool setup_internal_() override;
+  canbus::Error set_mode_(const CANCTRL_REQOP_MODE mode);
+
+  uint8_t read_register_(const REGISTER reg);
+  void read_registers_(const REGISTER reg, uint8_t values[], const uint8_t n);
+  void set_register_(const REGISTER reg, const uint8_t value);
+  void set_registers_(const REGISTER reg, uint8_t values[], const uint8_t n);
+  void modify_register_(const REGISTER reg, const uint8_t mask,
+                        const uint8_t data);
+
+  void prepare_id_(uint8_t *buffer, const bool ext, const uint32_t id);
+  canbus::Error reset_();
+  canbus::Error set_clk_out_(const CanClkOut divisor);
+  canbus::Error set_bitrate_(canbus::CanSpeed can_speed);
+  canbus::Error set_bitrate_(canbus::CanSpeed can_speed,
+                             const CanClock can_clock);
+  canbus::Error set_filter_mask_(const MASK num, const bool ext,
+                                 const uint32_t ul_data);
+  canbus::Error set_filter_(const RXF num, const bool ext,
+                            const uint32_t ul_data);
+  canbus::Error send_message_(const TXBn txbn,
+                              const struct canbus::CanFrame *frame);
+  canbus::Error send_message_(const struct canbus::CanFrame *frame); 
+  canbus::Error read_message_(const RXBn rxbn, struct canbus::CanFrame *frame);
+  canbus::Error read_message_(struct canbus::CanFrame *frame);
+  bool check_receive_();
+  bool check_error_();
+  uint8_t get_error_flags_();
+  void clearRXnOVRFlags();
+  uint8_t getInterrupts();
+  uint8_t getInterruptMask();
+  void clearInterrupts();
+  void clearTXInterrupts();
+  uint8_t get_status_();
+  void clearRXnOVR();
+  void clearMERR();
+  void clearERRIF();
```

Comment:
```suggestion
  void clear_errif_();
```
See contributing guide.

Reply:
changed all method names to required format